File: 06-spring/Spring-MVC/src/main/java/com/webby/model/Glue.java
package com.webby.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.SequenceGenerator;

@Entity
public class Glue {

	@Id
//	this does not work, there is a way to do a custom sequence though. just research
//	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "mySequence")
//	@SequenceGenerator(initialValue = 2, allocationSize = 25, name = "mySequence")
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;

	private String brand;

	private int stickyLevel;

	private boolean crazy;

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getBrand() {
		return brand;
	}

	public void setBrand(String brand) {
		this.brand = brand;
	}

	public int getStickyLevel() {
		return stickyLevel;
	}

	public void setStickyLevel(int stickyLevel) {
		this.stickyLevel = stickyLevel;
	}

	public boolean isCrazy() {
		return crazy;
	}

	public void setCrazy(boolean crazy) {
		this.crazy = crazy;
	}

	public Glue(int id, String brand, int stickyLevel, boolean crazy) {
		super();
		this.id = id;
		this.brand = brand;
		this.stickyLevel = stickyLevel;
		this.crazy = crazy;
	}

	public Glue() {
		super();
		// TODO Auto-generated constructor stub
	}

	@Override
	public String toString() {
		return "Glue [id=" + id + ", brand=" + brand + ", stickyLevel=" + stickyLevel + ", crazy=" + crazy + "]";
	}

}

File: 06-spring/Spring-Example/src/app/models/glue.ts
export class Glue {
  constructor(
    // tslint:disable: variable-name
    private _id: number,
    private _brand: string,
    private _stickyLevel: number,
    private _crazy: boolean
  ) {}
  get id(): number {
    return this._id;
  }
  set id(id: number) {
    this._id = id;
  }
  get brand(): string {
    return this._brand;
  }
  set brand(brand: string) {
    this._brand = brand;
  }
  get stickyLevel(): number {
    return this._stickyLevel;
  }
  set stickyLevel(sticky: number) {
    this._stickyLevel = sticky;
  }
  get crazy(): boolean {
    return this._crazy;
  }
  set crazy(crazy: boolean) {
    this._crazy = crazy;
  }
}
